zomes/service/code/src/invoice.rs

```diff
@@ -78,8 +86,64 @@ fn validate_invoiced_logs(context: hdk::EntryValidationData<InvoicedLogs>) -> Re
     }
 }
 
-pub fn handle_generate_invoice() -> ZomeApiResult<Address> {
-    //** First get the payment prefs
+fn is_unpaid(_invoiced_logs: InvoicedLogs) -> bool {
+    // Should bridge to Holofuel to get payment status
+    true
+}
+
+fn list_unpaid_invoices() -> Vec<Address> {
+    let invoices = match hdk::query("invoiced_logs".into(), 0, 0) {
```

Comment:
You can use hdk::query_results to get all the Entries and ChainHeaders, so that we can A) collect a date range for the requests, and B) avoid needing to get the logs entries separately.

Reply:
The logics for getting the `expiration date` is not set yet, so we could keep this for future releases?